esmf_regrid/schemes.py

```diff
@@ -2,8 +2,10 @@
 
 from collections import namedtuple
 import copy
+import functools
 
 import iris
+from iris._lazy_data import map_complete_blocks
```

Comment:
Is there any alternative to importing from a _ module (i.e. private) in iris? Not a deal breaker, these two libraries are very tightly coupled already.

Reply:
As far as I can tell, the alternative would be to effectively rewrite the function. I don't believe it's a very large or complicated function so it could be doable, but since this work is more or less inspired by the work on iris I've been keeping the links obvious for now. I imagine this may end up being replaced when optimising for performance eventually.

Reply:
No need to rewrite, this library is tightly coupled to iris so it makes sense to reuse the code over there.  We may want to move `map_complete_blocks` to the public api in iris if this becomes a regular thing we do.